File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        
        newStr = []
        for char in s:
            # print(char)
            if ord("a") <= ord(char) <= ord("z") or ord("A") <= ord(char) <= ord("Z"):
                newStr.append(char.lower())
            elif char in "0123456789":
                newStr.append(char)
        
        print(newStr)
        left, right = 0, len(newStr) - 1
        while left < right:
            if newStr[left] != newStr[right]:
                return False
            left += 1
            right -= 1
        
        return True

File: 0125-valid-palindrome/0125-valid-palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        
        StringBuilder stringBuilder = new StringBuilder();
        
        for (int i = 0; i < s.Length; i++) {
            char c = s[i];
            if (char.IsLetter(c)) {
                stringBuilder.Append(char.ToLower(c));
            } else if (char.IsDigit(c)) {
                stringBuilder.Append(char.ToLower(c));
            }
        }
        
        string s_copy = stringBuilder.ToString();
        int left = 0;
        int right = s_copy.Length - 1;
        while (left < right) {
            if (s_copy[left] != s_copy[right]) {
                return false;
            }
            left += 1;
            right -= 1;
        }
        
        
        return true;
    }
}